common/src/http.rs

```diff
@@ -21,14 +21,46 @@ mod icp {
 
     use super::HttpResponse;
 
+    // TODO: should this be configurable?
+    const MAX_REDIRECTS: usize = 2;
+
     pub async fn get(url: &str) -> Result<HttpResponse, HttpError> {
-        let req = CanisterHttpRequestArgument {
-            url: url.to_owned(),
-            method: HttpMethod::GET,
-            ..Default::default()
-        };
-        let resp = http_request(req, DEFAULT_HTTP_OUTCALL_COST).await?;
-        resp.0.try_into()
+        let mut url = url.to_owned();
+
+        for _ in 0..1 + MAX_REDIRECTS {
+            let req = CanisterHttpRequestArgument {
+                url: url.to_owned(),
+                method: HttpMethod::GET,
+                ..Default::default()
+            };
+            let resp = http_request(req, DEFAULT_HTTP_OUTCALL_COST).await?.0;
+
+            match http_status_from_nat(&resp.status)? {
+                // moved_permamently / found / see_other / temporary_redirect / permament_redirect
+                // redirect and retry
+                301..=303 | 307..=308 => {
+                    url = resp
+                        .headers
+                        .iter()
+                        .find_map(|header| {
+                            if header.name == "location" || header.name == "Location" {
```

Comment:
The HTTP standard says that [field names are case-insensitive](https://datatracker.ietf.org/doc/html/rfc2616#section-4.2). So use [eq_ignore_ascii_case](https://doc.rust-lang.org/stable/std/primitive.str.html#method.eq_ignore_ascii_case).

Reply:
cool, made it exactly to not allocate on to_lowercase, wasn't aware of that method